### PR TITLE
[framework] enable freezing of coin

### PR DESCRIPTION
**Pull request by davidiw**
some coins in an account may need to be frozen depending on their issuer requirements.
most coins won't need this, so make it easy to throw away the freeze
capability.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2912)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/aptos_coin.move [hunk 1]
```diff
@@ -34,18 +34,19 @@ module aptos_framework::aptos_coin {
...
             aptos_framework,
             string::utf8(b"Aptos Coin"),
             string::utf8(b"APT"),
             8, /* decimals */
             false, /* monitor_supply */
         );
 
-        (mint_cap, burn_cap)
+        coin::destroy_freeze_cap(freeze_cap);
+        (burn_cap, mint_cap)
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/aptos_coin.move:49]: Note, anyone else who uses the test signature will now be confused

File: aptos-move/framework/aptos-framework/sources/coin.move [hunk 2]
```diff
@@ -166,6 +173,8 @@ module aptos_framework::coin {
     /// Burn `coin` from the specified `account` with capability.
     /// The capability `burn_cap` should be passed as a reference to `BurnCapability<CoinType>`.
     /// This function shouldn't fail as it's called as part of transaction fee burning.
+    ///
+    /// Note: This bypasses CoinStore::frozen.
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/coin.move:177]: Maybe saying frozen coins can still be burned is more explanatory

File: aptos-move/framework/aptos-framework/sources/coin.move [hunk 3]
```diff
@@ -623,6 +667,56 @@ module aptos_framework::coin {
         destroy_zero(zero);
     }
 
+    #[test(account = @0x1)]
+    #[expected_failure(abort_code = 0x50009)]
+    public entry fun withdraw_frozen(account: signer) acquires CoinStore {
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/coin.move:672]: Maybe you want a test showing you can burn when frozen?

**Review by gregnazario — APPROVED** [review]
My only concern is you changed the order of mint and burn in the output

File: aptos-move/framework/aptos-framework/sources/aptos_coin.move (hunk 1 above)
Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/aptos_coin.move:49]: confused ... maybe but it is statically typed :)

**Pull request merged**